### PR TITLE
tilix: 1.8.5 -> 1.8.9

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
Update `tilix` to its latest version and build with meson. This required me to also bump `gtkd`.

This currently fails pretty early with
```
data/meson.build:99:0: ERROR:  Output file name must not contain a subdirectory.
```

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/tilix/default.nix [hunk 1]
```diff
@@ -1,33 +1,33 @@
...
+  postPatch = ''
+    chmod +x meson_post_install.py
+    patchShebangs meson_post_install.py
   '';
 
   postInstall = with gnome3; ''
-    ${glib.dev}/bin/glib-compile-schemas $out/share/glib-2.0/schemas
-
     wrapProgram $out/bin/tilix \
       --prefix LD_LIBRARY_PATH ":" "${libsecret}/lib"
```
Comment (jtojnar) [inline pkgs/applications/misc/tilix/default.nix]: Maebe use `gappsWrapperArgs`?

File: pkgs/applications/misc/tilix/no-dub.patch [hunk 2]
```diff
@@ -0,0 +1,26 @@
+diff --git a/meson.build b/meson.build
+index feff315..cb92631 100644
+--- a/meson.build
++++ b/meson.build
+@@ -93,12 +93,8 @@ source_root = meson.source_root()
+ sources_dir = include_directories('source/')
+ 
+ # Dependencies
+-# Note relying on dub to fetch and build dependencies before running meson build is temporary due to a bug in Meson 0.48
```
Comment (jtojnar) [inline pkgs/applications/misc/tilix/no-dub.patch]: I asked for clarification about the bug: 
https://github.com/gnunn1/tilix/commit/4092dfc7df6f8a2d2c90b29cf43995daac080a04#commitcomment-31853141

Comment (worldofpeace) [reply pkgs/applications/misc/tilix/no-dub.patch]: Didn't particularly look around to see if there was an issue with this, but I did see improvements for dlang hit in 0.49.0.

**Comment by worldofpeace** [comment]
@jtojnar See anything wrong here https://github.com/gnunn1/tilix/blob/54d37141a75945e291ea3051af9caa7bb943aa40/data/meson.build#L99 ?

**Comment by jtojnar** [comment]
Missing `install: true`?

**Comment by worldofpeace** [comment]
> Oh right, read the OP. Yeah, that is pretty annoying issue – meson simply does not allow that. I would suggest creating a `meson.build` file in the `dbus` dir, or maybe just change the output to `@BASENAME@` – it should not matter where it is generated, since we only need it for installation.


Thanks, I'll try that :+1:

**Comment by worldofpeace** [comment]
Hmm I guess it really must be broken judging from the build failure

```
tilix@exe/source_gx_i18n_l10n.d.o: In function `_D2gx4i18n4l10n1_FAyaZQe':
../source/gx/i18n/l10n.d:(.text._D2gx4i18n4l10n1_FAyaZQe[_D2gx4i18n4l10n1_FAyaZQe]+0x42): undefined reference to `_D4glib3StrQe9toStringzFNaAyaZPa'
../source/gx/i18n/l10n.d:(.text._D2gx4i18n4l10n1_FAyaZQe[_D2gx4i18n4l10n1_FAyaZQe]+0x56): undefined reference to `_D4glib3StrQe9toStringzFNaAyaZPa'
../source/gx/i18n/l10n.d:(.text._D2gx4i18n4l10n1_FAyaZQe[_D2gx4i18n4l10n1_FAyaZQe]+0x64): undefined reference to `c_g_dgettext'
../source/gx/i18n/l10n.d:(.text._D2gx4i18n4l10n1_FAyaZQe[_D2gx4i18n4l10n1_FAyaZQe]+0x71): undefined reference to `_D4glib3StrQe8toStringFNaPxamZAya'
tilix@exe/source_gx_i18n_l10n.d.o: In function `_D2gx4i18n4l10n2C_FAyaQdZQg':
../source/gx/i18n/l10n.d:(.text._D2gx4i18n4l10n2C_FAyaQdZQg[_D2gx4i18n4l10n2C_FAyaQdZQg]+0x5a): undefined reference to `_D4glib3StrQe9toStringzFNaAyaZPa'
../source/gx/i18n/l10n.d:(.text._D2gx4i18n4l10n2C_FAyaQdZQg[_D2gx4i18n4l10n2C_FAyaQdZQg]+0x6e): undefined reference to `_D4glib3StrQe9toStringzFNaAyaZPa'
../source/gx/i18n/l10n.d:(.text._D2gx4i18n4l10n2C_FAyaQdZQg[_D2gx4i18n4l10n2C_FAyaQdZQg]+0x82): undefined reference to `_D4glib3StrQe9toStringzFNaAyaZPa'
../source/gx/i18n/l10n.d:(.text._D2gx4i18n4l10n2C_FAyaQdZQg[_D2gx4i18n4l10n2C_FAyaQdZQg]+0x94): undefined reference to `c_g_dpgettext2'
../source/gx/i18n/l10n.d:(.text._D2gx4i18n4l10n2C_FAyaQdZQg[_D2gx4i18n4l10n2C_FAyaQdZQg]+0xa1): undefined reference to `_D4glib3StrQe8toStringFNaPxamZAya'
collect2: error: ld returned 1 exit status
Error: linker exited with status 1
ninja: build stopped: subcommand failed.
builder for '/nix/store/07ngw7s8bqlm6vgm5z1fhfsrdw0v70m8-tilix-1.8.9.drv' failed with exit code 1
error: build of '/nix/store/07ngw7s8bqlm6vgm5z1fhfsrdw0v70m8-tilix-1.8.9.drv' failed
```

**Comment by jtojnar** [comment]
Looks like glib is not linked, try adding `dependency('glib')` to https://github.com/gnunn1/tilix/blob/a6ea2cb08481710218de06bb55cc5d03ad3e1476/meson.build#L113

**Comment by worldofpeace** [comment]
Decided to switch to meson later in d4224f05074b6b8b44fd9bd68e12d4f55341b872, though I'd really like to put together an issue for meson as to why this isn't working.

**Pull request closed without merging**